core/tests/puppeteer-acceptance-tests/puppeteer-testing-utilities/puppeteer-utils.js

```diff
@@ -0,0 +1,238 @@
+        });
+      });
+
+    return this.page;
+  }
+
+  /**
+   * Function to sign in the user with the given email to the Oppia website.
+   * @param {string} email - The email of the user.
+   */
+  async signInWithEmail(email) {
+    await this.goto(testConstants.URLs.home);
+    if (!this.userHasAcceptedCookies) {
+      await this.clickOn('OK');
+      this.userHasAcceptedCookies = true;
+    }
+    await this.clickOn('Sign in');
+    await this.type(testConstants.SignInDetails.inputField, email);
+    await this.clickOn('Sign In');
+    await (this.page).waitForNavigation({waitUntil: 'networkidle0'});
+  }
+
+  /**
+   * This function signs up a new user with the given username and email.
+   * @param {string} userName - The username of the user.
+   * @param {string} signInEmail - The email of the user.
+   */
+  async signUpNewUser(userName, signInEmail) {
+    await this.signInWithEmail(signInEmail);
+    await this.type('input.e2e-test-username-input', userName);
+    await this.clickOn('input.e2e-test-agree-to-terms-checkbox');
+    await this.page.waitForSelector(
+      'button.e2e-test-register-user:not([disabled])');
+    await this.clickOn(LABEL_FOR_SUBMIT_BUTTON);
+    await (this.page).waitForNavigation({waitUntil: 'networkidle0'});
+  }
+
+  /**
+   * This function waits for a component using its CSS selector to load.
+   * @param {string} selector - The CSS selector of the component.
+   */
+  async waitForPageToLoad(selector) {
+    await this.page.waitForSelector(selector);
+  }
+
+  /**
+   * Function to reload the current page.
+   */
+  async reloadPage() {
+    await this.page.reload({waitUntil: ['networkidle0', 'domcontentloaded']});
+  }
+
+  /**
+   * The function clicks the element using the text on the button.
+   * @param {string} selector - The text on the button or the CSS selector of
+   * the element to be clicked
+   */
+  async clickOn(selector) {
+    try {
+      const [button] = await (this.page).$x(
+        '//' + '*[contains(text(), "' + selector + '")]');
+      await button.click();
+    } catch {
+      await (this.page).waitForSelector(selector);
+      await (this.page).click(selector);
+    }
+  }
+
+  /**
+   * The function to assign a role to a user.
+   * @param {string} username - The username to which role would be assigned.
+   * @param {string role - The role that would be assigned to the user.
+   */
+  async assignRoleToUser(username, role) {
+    await this.goto(rolesEditorTab);
+    await this.type(roleEditorInputField, username);
+    await this.clickOn(roleEditorButtonSelector);
+    await this.clickOn('Add role');
+    await this.clickOn(rolesSelectDropdown);
+    await this.page.evaluate(async(role) => {
+      const allRoles = document.getElementsByClassName('mat-option-text');
+      for (let i = 0; i < allRoles.length; i++) {
+        if (allRoles[i].innerText.toLowerCase() === role) {
+          allRoles[i].click({waitUntil: 'networkidle0'});
+          return;
+        }
```

Comment:
Probably should throw error if the role is not found -- that can help other devs using this function.

Reply:
done